File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        max_candy = -1
        for candy in candies:
            if candy > max_candy:
                max_candy = candy
        candy_list = []
        for candy in candies:
            if candy + extraCandies >= max_candy:
                candy_list.append(True)
            else:
                candy_list.append(False)
        return candy_list

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int maxCandy = 0;
        for(int i = 0; i < candies.length; i++) {
            if(maxCandy < candies[i]) {
                maxCandy = candies[i];
            }
        }
        List<Boolean> kidList = new ArrayList<>();
        for(int i = 0; i < candies.length; i++) {
            if(candies[i] + extraCandies >= maxCandy) {
                kidList.add(true);
            } else {
                kidList.add(false);
            }
        }
        return kidList; 
    }
}